game-core/src/test/java/org/triplea/live/servers/ServerYamlValidityTest.java

```diff
@@ -0,0 +1,98 @@
+package org.triplea.live.servers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.text.IsEmptyString.emptyString;
+import static org.triplea.test.common.StringToInputStream.asInputStream;
+import static org.triplea.test.common.TestDataFileReader.readContents;
+
+import feign.template.UriUtils;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.hamcrest.core.StringEndsWith;
+import org.junit.jupiter.api.Test;
+import org.triplea.util.Version;
+
+/**
+ * Checks the current 'servers.yml' to verify it is semantically valid. eg:
+ *
+ * <ul>
+ *   <li>Contains a latest version, greater than 0.0
+ *   <li>Inactive versions have null URI and no message
+ *   <li>Active versions have a URI and a message
+ *   <li>URI does not end with a trailing slash
+ *   <li>URI is absolute, it contains protocol and host
+ *   <li>No exact version duplicates
+ * </ul>
+ */
+class ServerYamlValidityTest {
+  private static final Supplier<InputStream> SERVERS_YML =
+      () -> asInputStream(readContents("servers.yml"));
+  private static final LiveServers liveServers = new ServerYamlParser().apply(SERVERS_YML.get());
+
+  @Test
+  void verifyLatestVersion() {
+    assertThat(liveServers.getLatestEngineVersion(), notNullValue());
+    assertThat(liveServers.getLatestEngineVersion().isGreaterThan(new Version("0.0")), is(true));
+  }
+
+  @Test
+  void uriValuesDoNotEndInSlash() {
+    liveServers.getServers().stream()
+        .map(ServerProperties::getUri)
+        .filter(Objects::nonNull)
+        .map(URI::toString)
+        .forEach(uriString -> assertThat(uriString, not(StringEndsWith.endsWith("/"))));
```

Comment:
Hmm isn't there an `assertAll` method?
In case there are no servers found, or filtered out completely this test wouldn't actually test anything, so at least testing the stream is not empty would be nice.
The same applies for all the other similar tests in this class

Reply:
Updated to add a check that we have at least one active server configured: ebc19b26d